File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/abstract_factory/IDepartment.java
package jv.com.tencent.cain.design_pattern.abstract_factory;

/**
 * @author : jiangyu
 * @date : 2020/3/20
 * @desc : xxx
 */
public interface IDepartment {
    /**
     * 插入数据
     * @param department
     */
    void insertDepartment(Department department);

    /**
     * 查询数据
     * @param id
     * @return
     */
    Department queryDepartment(int id);
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/abstract_factory/IDepartment.kt
package kt.com.tencent.cain.desigin_pattern.abstract_factory

/**
 * @author : jiangyu
 * @date   : 2020/3/20
 * @desc   : xxx
 */
interface IDepartment {
    /**
     * 插入数据
     */
    fun insertDepartment(department: Department)

    /**
     * 查询数据
     */
    fun queryDepartment(id: Int): Department
}